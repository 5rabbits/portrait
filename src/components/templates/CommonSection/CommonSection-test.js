import { React, toJSON, mount } from 'test/helper'
import CommonSection from 'templates/CommonSection'

describe('CommonSection', () => {
  it('should render with minimum props', () => {
    const component = mount(<CommonSection />)

    expect(toJSON(component)).toMatchSnapshot()
  })
})
