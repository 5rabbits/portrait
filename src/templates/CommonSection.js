import React, { PureComponent } from 'react'
import { Section } from '../main'
import { Offscreen, States } from '../shared'

export default class CommonSection extends PureComponent {
  buildContent() {
    if (!this.props.panel) {
      return this.buildStates()
    }

    return (
      <Offscreen
        panel={this.props.panel}
        content={this.buildStates()}
      />
    )
  }

  buildStates() {
    return (
      <States
        content={this.props.content}
        empty={this.props.empty}
      />
    )
  }

  render() {
    return (
      <Section
        controls={this.props.controls}
        content={this.buildContent()}
      />
    )
  }
}
