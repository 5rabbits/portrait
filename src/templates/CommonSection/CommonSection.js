import React, { PureComponent } from 'react'
import childrenPropType from 'propTypes/children'
import { Section } from 'main'
import { Offscreen, States } from 'shared'

export default class CommonSection extends PureComponent {
  static propTypes = {
    controls: childrenPropType,
    content: childrenPropType,
    empty: childrenPropType,
    panel: childrenPropType,
  }

  buildContent() {
    if (!this.props.panel) {
      return this.buildStates()
    }

    return (
      <Offscreen
        panel={this.props.panel}
        content={this.buildStates()}
      />
    )
  }

  buildStates() {
    return (
      <States
        content={this.props.content}
        empty={this.props.empty}
      />
    )
  }

  render() {
    return (
      <Section
        controls={this.props.controls}
        content={this.buildContent()}
      />
    )
  }
}
