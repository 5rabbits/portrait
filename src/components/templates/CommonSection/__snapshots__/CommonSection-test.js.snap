// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CommonSection should render with minimum props 1`] = `
<CommonSection>
  <Section
    toolbar={null}
  >
    <section
      className="Section"
    >
      <section
        className="Section__contents"
      >
        <States>
          <div
            className="section-states"
          >
            <article
              className="hidden section-state"
              id="block-content-loading"
            >
              <Loader>
                <EmptyView>
                  <div
                    className="empty-view"
                  >
                    <div
                      className="empty-container"
                    >
                      <div
                        className="empty-dialog"
                      >
                        <i
                          className="fa fa-circle-o-notch fa-5x color-soft fa-spin"
                        />
                      </div>
                    </div>
                  </div>
                </EmptyView>
              </Loader>
            </article>
            <article
              className="hidden section-state"
              id="block-content-empty"
            >
              <EmptyView>
                <div
                  className="empty-view"
                >
                  <div
                    className="empty-container"
                  >
                    <div
                      className="empty-dialog"
                    />
                  </div>
                </div>
              </EmptyView>
            </article>
            <article
              className="hidden- section-state"
              id="block-content-data"
            />
          </div>
        </States>
      </section>
    </section>
  </Section>
</CommonSection>
`;
